src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -899,6 +1042,209 @@ client_session* get_session(document::view op, entity::map& map) {
+            options.hint(hint{arguments["hint"].get_string().value});
+        else
+            options.hint(hint{arguments["hint"].get_document().value});
+    }
+    if (arguments["upsert"]) {
+        options.upsert(arguments["upsert"].get_bool().value);
+    }
+
+    std::int32_t matched_count = 0;
+    bsoncxx::stdx::optional<std::int32_t> modified_count;
+    std::int32_t upserted_count = 0;
+    bsoncxx::stdx::optional<types::bson_value::view> upserted_id{};
+
+    auto update_many_result = coll.update_many(filter, update, options);
+    if (update_many_result) {
+        matched_count = update_many_result->matched_count();
+
+        // Server versions below 2.4 do not return an `nModified` count.
+        try {
+            modified_count = update_many_result->modified_count();
+        } catch (const std::exception& e) {
+        }
+
+        upserted_count = update_many_result->result().upserted_count();
+
+        if (auto upserted_element = update_many_result->upserted_id()) {
+            upserted_id = upserted_element->get_value();
+        }
+    }
+
+    auto result = builder::basic::document{};
+    result.append(builder::basic::kvp(
+        "result",
+        [matched_count, modified_count, upserted_id](builder::basic::sub_document subdoc) {
+            subdoc.append(builder::basic::kvp("matchedCount", matched_count));
+
+            if (modified_count) {
+                subdoc.append(builder::basic::kvp("modifiedCount", *modified_count));
+            }
+
+            if (upserted_id) {
+                subdoc.append(builder::basic::kvp("upsertedId", *upserted_id));
+            }
+        }));
+
+    return result.extract();
+}
+
+document::value count_documents(collection& coll, document::view operation) {
+    document::view arguments = operation["arguments"].get_document().value;
+    document::value empty_filter = builder::basic::make_document();
+    document::view filter;
+    if (arguments["filter"]) {
+        filter = arguments["filter"].get_document().value;
+    } else {
+        filter = empty_filter.view();
+    }
+    options::count options{};
+
+    if (arguments["collation"]) {
+        options.collation(arguments["collation"].get_document().value);
+    }
+    if (arguments["limit"]) {
+        options.limit(as_int64(arguments["limit"]));
+    }
+    if (arguments["skip"]) {
+        options.skip(as_int64(arguments["skip"]));
+    }
+    if (arguments["hint"]) {
+        if (arguments["hint"].type() == bsoncxx::v_noabi::type::k_string)
+            options.hint(hint{arguments["hint"].get_string().value});
+        else
+            options.hint(hint{arguments["hint"].get_document().value});
+    }
+
+    int64_t count = coll.count_documents(filter, options);
+    auto result = builder::basic::document{};
+    result.append(builder::basic::kvp("result", [count](builder::basic::sub_document subdoc) {
+        subdoc.append(builder::basic::kvp("count", count));
+    }));
```

Comment:
I build a document `{"result": {"count": count}}` here and in `estimated_document_count` instead of returning an `int64_t` since `operations::run` always returns a `document::value`.